### PR TITLE
Fix http2 concurrent bug

### PR DESCRIPTION
**Pull request by taoyuanyuan**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Sign the CLA
Make sure you have signed the [CLA](https://www.clahub.com/agreements/alipay/sofa-mosn)

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, 
details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases.
And you need to show the ut result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/proxy/retrystate.go [hunk 1]
```diff
@@ -91,28 +106,33 @@ func (r *retryState) doRetryCheck(headers types.HeaderMap, reason types.StreamRe
...
 
 	if r.retryOn {
-		// TODO: add retry policy to decide retry or not. use default policy now
-		if headers != nil {
-			// default policy , mapping all headers to http status code
-			code, err := protocol.MappingHeaderStatusCode(r.upstreamProtocol, headers)
-			if err == nil {
-				// todo: support config?
-				return code >= http.InternalServerError
+		if r.retryConnFailed {
```
Comment (nejisama) [inline pkg/proxy/retrystate.go]: 复用retryOn
